### PR TITLE
completed three points from homework

### PR DESCRIPTION
**Pull request by iq100bl**
внизу Readme по программе. там пример того, как я вводил данные. потестить другой вид форматирования неуспел. комментарии потом удалю. были как стимул найти

### CODE REVIEW

**Comment by iq100bl** [comment]
Докидывал всё в кучу, хочу посмотреть как будут мержиться комиты друг с другомю если я правильно помню, должен быть конфликт? преобразовал всё как мог. в 3 комите избавился от всех методов с тремя параметрами. все что повторяется я поправлю сам, нужно время=)

File: Diary/Diary/DiaryService.cs [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Diary
+{
+    internal class DiaryService
+    {
+        static readonly PriorityTask[] weeklyTasks = new PriorityTask[100];
+        internal static int RecodrTask(int numberTask)
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:12]: * empty line
* missing access modifier on `weeklyTasks`
* you use the highest type `PriorityTask` to store tasks?
* RecodrTask -> RecordTask (install spell checker)
* bad design - `numberTask` should not be returned, if you need an actual count create property
```c#
public int Count => weeklyTasks.Length;
```

File: Diary/Diary/DiaryService.cs [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+            for (int i = 1; i < numberTask + 1; i++)
+            {
+                Console.WriteLine("task " + i + ": "
+                                   + weeklyTasks[i - 1].name
+                                   + " "
+                                   + CheckData(i)
+                                   + " "
+                                   + CheckTime(i)
+                                   + " "
+                                   + weeklyTasks[i - 1].priority);
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:33]: looks quite ugly, don't format code like this. Check this way:
```c#
var date = GetDate()
var time = GetTime();
var name = ....
Console.WriteLine($"... {date} ... {time} ... {name}");
```

File: Diary/Diary/DiaryService.cs [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+            }
+        }
+
+        internal void FiltrTask(int numberTask)
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:37]:
```suggestion
        internal void FilterTask(int taskNumber)
```

File: Diary/Diary/DiaryService.cs [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+        {
+            Console.Write("By date or by priority?");
+            string filtr = Console.ReadLine();
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:40]:
```suggestion
            string filter = Console.ReadLine();
```

File: Diary/Diary/DiaryService.cs [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+
+            switch (filtr.ToLower())
+            {
+                case "date":
+                    FiltrToDate(numberTask);
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:45]:
```suggestion
                    FlterByDate(numberTask);
```

File: Diary/Diary/DiaryService.cs [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+                    break;
+
+                case "priority":
+                    FiltrToPriority(numberTask);
+                    break;
+
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:51]: here we don't need an empty space 😄

File: Diary/Diary/DiaryService.cs [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+                default:
+                    Console.WriteLine("incorrect parameter format");
+                    break;
+            }
+        }
+
+        private static void Records1Parametr(string[] taskParametrs, int numberTask)
+        {
+            PriorityTask diaryTask = new(taskParametrs[0]);
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:102]: something wrong with the inheritance model, this is not a priority task. priority task must have a priority provided

File: Diary/Diary/DiaryService.cs [hunk 8]
```diff
@@ -0,0 +1,220 @@
...
+            weeklyTasks[numberTask] = diaryTask;
+        }
+
+        private static void Records2Parametrs(string[] taskParametrs, int numberTask)
+        {
+            PriorityTask diaryTask = new(taskParametrs[0],
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:108]: formatting

File: Diary/Diary/DiaryService.cs [hunk 9]
```diff
@@ -0,0 +1,220 @@
...
+                                                      DateTime.Parse(taskParametrs[1]),
+                                                      DateTime.Parse(taskParametrs[2]));
+            weeklyTasks[numberTask] = diaryTask;
+            diaryTask.GetAlarm();
+        }
+
+        private static void Record4Parametrs(string[] taskParametrs, int numberTask)
+        {
+            PriorityTask diaryTask = new(taskParametrs[0],
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:125]: don't format like that, create variables if a line is too long

File: Diary/Diary/DiaryService.cs [hunk 10]
```diff
@@ -0,0 +1,220 @@
...
+                                                                          DateTime.Parse(taskParametrs[2]),
+                                                                          taskParametrs[3]);
+            weeklyTasks[numberTask] = diaryTask;
+            diaryTask.GetAlarm();
+        }
+
+
+        private static string CheckTime(int i)
+        {
+            if (weeklyTasks[i - 1].timeTask == default(DateTime))
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:136]: `i` means index, why do you subtract 1? That means it is not index it is a task number. Name the parameter accordingly.

File: Diary/Diary/DiaryService.cs [hunk 11]
```diff
@@ -0,0 +1,220 @@
...
+            {
+                return " ";
+            }
+            return weeklyTasks[i - 1].timeTask.ToLongTimeString();
+        }
+
+        private static string CheckData(int i)
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:143]: you probably mean 
```suggestion
        private static string CheckDate(int i)
```

File: Diary/Diary/DiaryService.cs [hunk 12]
```diff
@@ -0,0 +1,220 @@
...
+            var x = 0;
+
+            do
+            {
+                if (weeklyTasks[i].priority == filtr)
+                {
+                    filtrTask[x] = weeklyTasks[i];
+                    x++;
+                }
+                i++;
```
Comment (rubleuskia) [inline Diary/Diary/DiaryService.cs:168]: too many iterators IMO, you can show tasks one by one or use advanced collections like `List` to collect the result.

File: Diary/Diary/PriorityTask.cs [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+namespace Diary
+{
+    internal class PriorityTask : RegularTask
+    {
+        internal string priority;
+
+        internal PriorityTask(string _name) : base(_name)
+        {
+            name = _name;
```
Comment (rubleuskia) [inline Diary/Diary/PriorityTask.cs:15]: this is done by base class, remove this line, same below

File: Diary/Diary/PriorityTask.cs [hunk 14]
```diff
@@ -0,0 +1,60 @@
...
+            dataTask = _dataTask;
+            timeTask = _timeTask;
+            priority = _priority;
+        }
+
+        internal override string GetAlarm()
+        {
+            return ToString();
```
Comment (rubleuskia) [inline Diary/Diary/PriorityTask.cs:41]: `ToString` is supposed to just print task, like "<name>, <date> <time>", but this method should calculate the days diff. 
Now everything is mixed up.

File: Diary/Diary/PriorityTask.cs [hunk 15]
```diff
@@ -0,0 +1,60 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Diary
+{
+    internal class PriorityTask : RegularTask
+    {
+        internal string priority;
```
Comment (rubleuskia) [inline Diary/Diary/PriorityTask.cs:11]: fields must be private

File: Diary/Diary/Program.cs [hunk 16]
```diff
@@ -0,0 +1,34 @@
+﻿using System;
+namespace Diary
+{
+     class Program
+    {
+        private static readonly DiaryService service = new();
+        static void Main(string[] args)
+        {
+            var numberTask = 0;
```
Comment (rubleuskia) [inline Diary/Diary/Program.cs:9]: remove this from here, there is no need to keep track of tasks number in the main program.

File: Diary/Diary/WeeklyTask.cs [hunk 17]
```diff
@@ -0,0 +1,17 @@
...
+namespace Diary
+{
+    abstract public class WeeklyTask
+    {
+        internal string name;
+        internal abstract string GetAlarm();
+        internal WeeklyTask(string name)
+        {
+        }        
+    }
```
Comment (rubleuskia) [inline Diary/Diary/WeeklyTask.cs:16]: fields must be private
empty lines
reorder methods

```suggestion
        private string name;
      
        internal WeeklyTask(string name)
        {
        }        
        
        internal abstract string GetAlarm();
    }
```

File: Diary/Diary/RegularTask.cs [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+        internal RegularTask(string _name, DateTime _dataTask, DateTime _timeTask) :base(_name)
+        {
+            name = _name;
+            dataTask = _dataTask;
+            timeTask = _timeTask;
+        }
+        internal override string GetAlarm()
+        {
+            return ToString();
+        }
```
Comment (rubleuskia) [inline Diary/Diary/RegularTask.cs:26]: fields must be private
empty lines
don't set data twice
```suggestion
        private DateTime dataTask;
        private DateTime timeTask;
        
        internal RegularTask(string _name) : base(_name)
        {
        }
        
        internal RegularTask(string _name, DateTime _dataTask, DateTime _timeTask) :base(_name)
        {
            dataTask = _dataTask;
            timeTask = _timeTask;
        }
        
        internal override string GetAlarm()
        {
            return ToString();
        }
```

**Pull request merged**